features/netsocket/SocketAddress.cpp

```diff
@@ -122,6 +122,7 @@ static void ipv6_from_address(uint8_t *bytes, const char *addr)
     // Move suffix to end
     memmove(&shorts[NSAPI_IPv6_BYTES/2-suffix], &shorts[0],
             suffix*sizeof(uint16_t));
+    memset(&shorts[0], 0, suffix*sizeof(uint16_t));
```

Comment:
If suffix is long memsetting suffix size can overwrite the beginning of memmoved suffix e.g.
if address is 2001::db8:1111:2222:3333:4444